providers/account.rb

```diff
@@ -234,3 +239,23 @@ def group_resource(exec_action)
       new_resource.updated_by_last_action(true) if r.updated_by_last_action?
   end
 end
+
+def keypair_resource(exec_action)
+  new_resource.ssh_keypair.each do |name, key|
+    # avoid variable scoping issues in resource block
+    key_name, key_content = name, key
+
+    home = Etc.getpwnam(new_resource.username).dir
```

Comment:
I suppose that my implementation will have the same issue. Is there a dynamic way of fixing this (without requiring a manual `group` attribute to be set)?